src/lib/response-triggers/help.ts

```diff
@@ -23,82 +23,235 @@ export default class Help extends Trigger {
 	}
 
 	isToTriggerOnPM(message : MessageObject) : boolean {
-		return message.text.toLowerCase().trim() === this.commandText;
+		return message.text.toLowerCase().trim().startsWith(Command.Help);
 	}
 
 	async createMessage(message : MessageObject) : Promise<MessageObject> {
-		let keywordsRequired = this.config.getConfig('requireValues');
-		let newMessage = `## What I can do
+		const pattern = new RegExp('^' + constants.botName, 'ui');
+		const helpPattern = new RegExp('^' + this.helpCommand, 'ui');
+		const command = message.text.trim().replace(pattern, '').trim().replace(helpPattern, '').trim();
+		let newMessage: string;
 
-* Give a peg 🎁!
-	1. To give someone a peg type: \`@${constants.botName} peg @bob {comment}\`.\n`;
+		if(!command) {
+			newMessage = this.createCommandListMessage(message);
+		} else {
+			switch(command.toLowerCase()){
+				case Command.Peg:
+					newMessage = this.createPegHelpMessage();
+					break;
+				case Command.Status:
+					newMessage = this.createStatusHelpMessage();
+					break;
+				case Command.Keywords:
+					newMessage = this.createKeywordsHelpMessage();
+					break;
+				case Command.Ping:
+					newMessage = this.createPingHelpMessage();
+					break;
+				case Command.Welcome:
+					newMessage = this.createWelcomeHelpMessage();
+					break;
+				case Command.Rotation:
+					newMessage = this.createRotationHelpMessage();
+					break;
+				case Command.Winners:
+					newMessage = this.createWinnersHelpMessage(message);
+					break;
+				case Command.Results:
+					newMessage = this.createResultsHelpMessage(message);
+					break;
+				case Command.Reset:
+					newMessage = this.createResetHelpMessage(message);
+					break;
+				case Command.Update:
+					newMessage = this.createUpdateHelpMessage(message);
+					break;
+				case Command.Finish:
+					newMessage = this.createFinishHelpMessage(message);
+					break;
+				case Command.NumberConfig:
+					newMessage = this.createNumberConfigHelpMessage(message);
+					break;
+				case Command.StringConfig:
+					newMessage = this.createStringConfigHelpMessage(message);
+					break;
+				case Command.RoleConfig:
+					newMessage = this.createRoleConfigHelpMessage(message);
+					break;
+				default:
+					newMessage = this.createDefaultHelpMessage();
+					break;
+			}
+		}
```

Comment:
Maybe refactor all of this out in to its own function, then call `let newMessage = thatFunction(message, command)`.

Reply:
Done

Reply:
If you put the `if(!command)` case in to the function as well, you can remove the null declaration of `newMessage`.

Reply:
Good call. Done